### ISSUE TITLE
Support `properties` + `anyOf` together

### ISSUE THREAD

**Issue by bradzacher** [bcherny/json-schema-to-typescript#243]
#229 and #167 both proposed this, but were closed even though it is not supported - the workaround in #167 was to change the schema, which I can't do because I don't control the schemas.

```js
{
  type: 'object',
  properties: {
    a: { type: 'string' },
    b: { type: 'number' }
  },
  anyOf: [
    { required: ['a'] },
    { required: ['b'] }
  ]
}
```

Should produce:

```ts
type T = 
  | {
    a: string,
    b?: number,
  }
  | {
    a?: string,
    b: number,
  }
```

**Comment by awlayton** [comment]
Changing the schema is also not a good workaround for me.

**Comment by awlayton** [comment]
It's a bit like cheating, but could this not be solved with a normalizer rule for `anyOf`/`allOf` etc. which merges in the base schema into all of the subschemas give to the boolean logic keyword?

As in, the above example would get normalized to this:
```js
{
  type: 'object',
  properties: {
    a: { type: 'string' },
    b: { type: 'number' }
  },
  anyOf: [
    {
      type: 'object',
      properties: {
        a: { type: 'string' },
        b: { type: 'number' }
      },
      required: ['a']
    },
    {
      type: 'object',
      properties: {
        a: { type: 'string' },
        b: { type: 'number' }
      },
      required: ['b']
    },
  ]
}
```

**Comment by awlayton** [comment]
My idea produces the desired output:
```ts
/* tslint:disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export type Foo =
  | {
      a: string;
      b?: number;
      [k: string]: unknown;
    }
  | {
      a?: string;
      b: number;
      [k: string]: unknown;
    };
```
I could write a normalizer for this and make a PR, but only you're willing to merge this idea in.

**Comment by carlosvfernandez** [comment]
I'm having the same problem... any fix/response?

**Comment by bcherny** [comment]
Quick update: I've been working on this in my spare time. It needs a bit more work before landing, to get the tests to pass: https://github.com/bcherny/json-schema-to-typescript/commit/2163b898e099c919b9090ab1093226bdd143cc73

**Comment by bcherny** [comment]
Fixed in 4aabd23, shipping as part of v10.
